crates/ethereum/Cargo.toml

```diff
@@ -8,20 +8,20 @@ rust-version = "1.62"
 
 [dependencies]
 anyhow = { workspace = true }
-async-trait = "0.1.59"
-ethers = "1.0.2"
-futures = { version = "0.3", default-features = false, features = ["std"] }
-lazy_static = "1.4.0"
+const-decoder = "0.3.0"
 pathfinder-common = { path = "../common" }
 pathfinder-retry = { path = "../retry" }
+primitive-types = { version = "0.12.1", features = ["serde"] }
```

Comment:
Should we just make our own `H256` and `H128` types? Shouldn't be so complex, or how heavy is this dep? Looks like many proc macros under the hood. But maybe its not that important.

Reply:
The `primitive-types` is a tiny part of `ethers`, so it is still less heavy that full `ethers`. But it is certainly doable, I will make a follow up for that just not to bloat this PR. https://github.com/eqlabs/pathfinder/issues/1035